src/main/java/tp/cap5buddy/logic/parser/Tokenizer.java

```diff
@@ -19,6 +23,112 @@ public Tokenizer(String input) {
+     *
+     * @param userInput User input
+     * @param prefixes Array of prefixes
+     * @return String array containing command arguments
+     * @throws ParseException If the user input could not be parsed.
+     */
+    public String[] tokenize(String userInput, Prefix... prefixes) throws ParseException {
+        List<PrefixPosition> prefixPositions = findAllPrefixPositions(userInput, prefixes);
+        return extractArguments(userInput, prefixPositions);
+    }
+
+    /**
+     *
+     *
+     * @param userInput User input
+     * @param prefixes Array of prefixes
+     * @return List of prefix positions
+     * @throws ParseException If the input could not be parsed.
+     */
+    public List<PrefixPosition> findAllPrefixPositions(String userInput, Prefix... prefixes) throws ParseException {
+        List<PrefixPosition> prefixPositions = new ArrayList<>();
+        for (Prefix prefix : prefixes) {
+            PrefixPosition prefixPosition = findPrefixPosition(userInput, prefix);
+            if (prefixPosition != null) {
+                prefixPositions.add(prefixPosition);
+            }
+        }
+        return prefixPositions;
+    }
+
+    /**
+     * Finds the position of the prefix in the user input.
+     *
+     * @param userInput User input.
+     * @param prefix Prefix.
+     * @return Prefix position.
+     * @throws ParseException If the same prefix is used more than once by the user.
+     */
+    public PrefixPosition findPrefixPosition(String userInput, Prefix prefix) throws ParseException {
+        String prefixSearch = " " + prefix.getPrefix();
+        int prefixIndex = userInput.indexOf(prefixSearch);
+
+        boolean hasMultipleSamePrefixes = hasMultipleSamePrefixes(userInput, prefix, prefixIndex);
+        if (hasMultipleSamePrefixes) {
+            String error = "User input has multiple arguments for the same prefix";
+            throw new ParseException(error);
+        }
+        return (prefixIndex == -1 ? null : new PrefixPosition(prefixIndex + 1, prefix));
+    }
+
+    /**
+     * Determines if the same prefix is used more than once.
+     *
+     * @param userInput User input.
+     * @param prefix Prefix.
+     * @param currentPrefixIndex Index of the current prefix in the user input.
+     * @return boolean.
+     */
+    public boolean hasMultipleSamePrefixes(String userInput, Prefix prefix, int currentPrefixIndex) {
+        int nextPrefixIndex = userInput.indexOf(" " + prefix.getPrefix(), currentPrefixIndex + 1);
+        boolean hasMultipleSamePrefix = nextPrefixIndex != -1;
+        return hasMultipleSamePrefix;
+    }
+
+    /**
+     * Extracts the command arguments using the list of prefix positions.
+     *
+     * @param userInput User input.
+     * @param prefixPositions Prefix positions.
+     * @return String array containing the command arguments.
+     */
+    public String[] extractArguments(String userInput, List<PrefixPosition> prefixPositions) {
```

Comment:
change to private, and change the name of the method, very similar to the one below